From eeef12c514269092ce679bf618a219fd8ad2aa68 Mon Sep 17 00:00:00 2001
From: Sascha Hlusiak <Sascha@hlusiak.de>
Date: Fri, 27 Mar 2009 11:14:00 -0700
Subject: [PATCH 07/23] iptunnel: allow ISATAP with stateless autoconf

please commit my patch below to the iproutes package. It just an incorrect
check so that adding an isatap tunnel with remote works, since it's needed if
one wants to use stateless autoconf. The current check makes tunnel mode
isatap unusable for all client users.
---
 ip/iptunnel.c |    7 +------
 1 files changed, 1 insertions(+), 6 deletions(-)

diff --git a/ip/iptunnel.c b/ip/iptunnel.c
index 0d9a17f..7bee162 100644
--- a/ip/iptunnel.c
+++ b/ip/iptunnel.c
@@ -252,13 +252,8 @@ static int parse_args(int argc, char **argv, int cmd, struct ip_tunnel_parm *p)
 		fprintf(stderr, "Broadcast tunnel requires a source address.\n");
 		return -1;
 	}
-	if (isatap) {
-		if (p->iph.daddr) {
-			fprintf(stderr, "no remote with isatap.\n");
-			return -1;
-		}
+	if (isatap)
 		p->i_flags |= SIT_ISATAP;
-	}
 
 	return 0;
 }
-- 
1.6.3.3

